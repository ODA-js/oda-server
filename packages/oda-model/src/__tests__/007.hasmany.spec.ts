import 'jest';
import {
  HasMany,
  hasManyDefaultMetaInfo,
  hasManyDefaultInput,
} from '../hasmany';
import { merge } from 'lodash';

describe('haOne', () => {
  it('default', () => {
    const res = new HasMany({ hasMany: 'A#' });
    expect(res.metadata).toMatchObject(hasManyDefaultMetaInfo);
    expect(res.toObject()).toMatchObject(
      merge({ hasMany: 'id@A#id' }, hasManyDefaultInput),
    );
    expect(res.modelType).toBe('HasMany');
    expect(res.verb).toBe('HasMany');
    expect(res.toObject()).toMatchSnapshot('toObject');
  });
});
