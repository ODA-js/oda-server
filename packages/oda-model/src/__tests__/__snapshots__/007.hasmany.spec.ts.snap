// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`haOne default: toObject 1`] = `
Object {
  "entity": undefined,
  "field": undefined,
  "hasMany": "id@A#id",
  "metadata": Object {
    "name": Object {
      "full": "HasMany",
      "normal": "",
      "short": "",
    },
    "persistence": Object {
      "embedded": false,
      "single": false,
      "stored": false,
    },
    "verb": "HasMany",
  },
  "name": "HasMany",
  "opposite": undefined,
}
`;
